### ISSUE TITLE
issues reading wavelength from TIF

### ISSUE THREAD

**Issue by DeboraOlivier** [macronucleus/chromagnon#23]
![image](https://user-images.githubusercontent.com/52743133/64430308-4fa0aa00-d0b8-11e9-8ce8-09bd9f763e5d.png)
Hello,

I have separate reference images for the color pairs 
- 488-405 (DAPI + 488 bleedthrough)
- 488-549: dual-labelled antibodies
- 549-647: dual-labelled antibodies
I want to make a single chromagnon file for all the different shifts 405-488-549-647.

To try do this, I saved a target file (original 4 channels) into 2 channels only in fiji using channel tools >split channels> merge relevant channels>save as tif.
when I put it into chromagnon, the program doesn't recognize the wavelength information and doesnt run.

How can I force the program or tell it the identity of the wavelengths? Is there a way to edit this information?
Is it possible to use different reference files to generate a single correction file (and then apply to 4-channel images)?
Thanks in advance

Debora

**Comment by macronucleus** [comment]
Good question!
I think the ImageJ file format does not contain wavelength information.

I guess combining channels using software provided by the microscope company should keep the wavelength information.
But still, you have to combine the alignment parameters.

What you can do at this moment is to edit “.chromagnon.csv” file using a spread sheet program to manually combine, and re-order the wavelength as in the four channel image.
You have to subtract by the values of the reference channel for translation and rotation, and divide for magnification.
If the four channel image was created by ImageJ, then the wavelengths must be also changed to 0, 1, 2, 3 in the order of channels in the image file.

**Comment by macronucleus** [comment]
Good question!
I think the ImageJ file format does not contain wavelength information.

Splitting channels using software provided by the microscope company should keep the wavelength information.

If you need to use ImageJ, then at this moment, what you can do is first measuring chromatic shifts in the reference image by Chromagnon to make a “chromagnon.csv” file. 
Then you have to manually edit “.chromagnon.csv” file using a spread sheet program or on Chromagon viewer.
The values in the column “wavelength” should be changed to 0, 1 in the order of channels in the image file.

Sorry for a clumsy solution.
I will consider a smarter solution to edit wavelengths and combine multiple reference files.

**Comment by DeboraOlivier** [comment]
hi,following up on this (better late than never). I have modified the CSV so taht the wavelengths name fit that of the SIR.dv file.
however, when I drag and drop in chromagnon v086, it doesn't work.
What am I doing wrong? 
I acquire teh DAPI signal in the channels 527 / 603 and 679 (and sometimes also in 403).
Reconstruct with Softworx 
then try to align
Any help will be appreciated
[chromagnon_git.zip](https://github.com/macronucleus/Chromagnon/files/5977894/chromagnon_git.zip)

**Comment by macronucleus** [comment]
Thank you for coming back to this issue.
The chromagnon file seems okay to me.
What does it happen if you try to align the target file with this file?

**Comment by DeboraOlivier** [comment]
Hi
no the issue is that I can't drag/drop or open the CSV reference file via
"References file" at all...DAPI_SIR_488-560-640.tif.chromagnon.csv
I have Chromagnon v0.86 on windows 10
I have no clue why it doesnt like the csv - am trying to add java just in
case

**Comment by DeboraOlivier** [comment]
update: even after Java JDK install, I can't open the csv file...

**Comment by macronucleus** [comment]
That's right, there are dialects in the csv format (including Microsoft Excel).
Please use Chromagnon v0.87 that I just uploaded in the release.
Thank you.

**Comment by DeboraOlivier** [comment]
it seems to work now! cool!
thanks
